Vineyard/Vineyard/Person.swift

```diff
@@ -6,17 +6,31 @@
 //
 
 import Foundation
+import FirebaseFirestore
 
 struct Person: Identifiable, Codable {
-    var id: String = UUID().uuidString
+    @DocumentID var id: String? = UUID().uuidString
     var name: String
-    var groups: [String] = []
-    var allProgress: [Progress] = []
     var email: String
-    var badges: [Badge] = []
+    var groupIDs: [String] = []
+    var allProgressIDs: [String] = []
+    var badgeIDs: [String] = []
+    
+    var groups: [Group]? = []
+    var allProgress: [Progress]? = []
+    var badges: [Badge]? = []
+    
+    enum CodingKeys: String, CodingKey {
+        case id
+        case name
+        case email
+        case groupIDs
+        case allProgressIDs
+        case badgeIDs
+    }
     
     mutating func addGroup(_ group: Group) {
-        self.groups.append(group.id)
+        self.groupIDs.append(group.id ?? UUID().uuidString)
```

Comment:
I would use a `guard` here and make sure the `id` is present instead of generating a new `id`.

```swift
mutating func addGroup(_ group: Group) {
    guard let id = group.id else { fatalError(/* ... */) }

    self.groupIDs.append(id)
}
```